Fitness/Controllers/HomeController.swift

```diff
@@ -171,45 +206,53 @@ extension HomeController: UICollectionViewDataSource {
             gymCell.setGymStatus(isOpen: gym.isOpen)
             gymCell.setGymHours(hours: getHourString(gym: gym))
             return gymCell
-        case .todaysClasses:
-            // swiftlint:disable:next force_cast
-            let todayClassesCell = collectionView.dequeueReusableCell(withReuseIdentifier: TodaysClassesCell.identifier, for: indexPath) as! TodaysClassesCell
-            todayClassesCell.collectionView.dataSource = self
-            todayClassesCell.collectionView.delegate = self
-            return todayClassesCell
         case .lookingFor:
             // swiftlint:disable:next force_cast
             let categoryCell = collectionView.dequeueReusableCell(withReuseIdentifier: CategoryCell.identifier, for: indexPath) as! CategoryCell
             categoryCell.title.text = tags[indexPath.row].name
-
+            
             let url = URL(string: tags[indexPath.row].imageURL)
             categoryCell.image.kf.setImage(with: url)
             return categoryCell
+        case .pros:
+            let discoverPros = collectionView.dequeueReusableCell(withReuseIdentifier: DiscoverProsCell.identifier, for: indexPath) as! DiscoverProsCell
+            discoverPros.collectionView.dataSource = self
+            discoverPros.collectionView.delegate = self
+            return discoverPros
+        case .todaysClasses:
+            // swiftlint:disable:next force_cast
+            let todayClassesCell = collectionView.dequeueReusableCell(withReuseIdentifier: TodaysClassesCell.identifier, for: indexPath) as! TodaysClassesCell
+            todayClassesCell.collectionView.dataSource = self
+            todayClassesCell.collectionView.delegate = self
+            return todayClassesCell
```

Comment:
the setting of `collectionView.dataSource` and `collectionView.delegate` should be done inside the cell class itself (in this case `DiscoverProsCell` & `TodaysClassesCell`). The dataSource and delegate should also be those cells too instead of being `HomeController`